### ISSUE TITLE
Unable to use SSL on Database Connections

### ISSUE THREAD

**Issue by shrock-n-awe** [ConfigHubPub/ConfigHubPlatform#36]
# ConfigHub.properties
```
# Settings for the primary ConfigHub Database.
db.main.host = mydb.mysql.database.azure.com
db.main.port = 3306
db.main.name = config-hub
db.main.username = confighub
db.main.password = pwd
db.main.useSSL = true

# Settings for the database that will store all incoming client (API) requests.
db.api.host = mydb.mysql.database.azure.com
db.api.port = 3306
db.api.name = config-hub-client-requests
db.api.username = confighub
db.api.password = pwd
db.api.useSSL = true

# Path to the location where all ConfigHub service logs are stored.
confighub.logging.path = /var/log/confighub

# Memory assigned to the ConfigHub service.  It is recommended to assign 4g or more.
confighub.memory = 8g

# HTTP and HTTPs ports
confighub.http.port = 80
confighub.https.port = 443

# Specify an override to the default self-signed certificate/keystore.
confighub.https.keystoreFile = cert/confighub_default.jks
confighub.https.keystoreAlias = confighub
confighub.https.keystorePass = confighub
```

# /var/log/confighub/configHub.log
```
2017-07-27 12:48:05,267 INFO HHH000204: Processing PersistenceUnitInfo [
	name: ConfigHubMain
	...]
2017-07-27 12:48:05,339 INFO HHH000412: Hibernate Core {5.1.1.Final}
2017-07-27 12:48:05,340 INFO HHH000206: hibernate.properties not found
2017-07-27 12:48:05,341 INFO HHH000021: Bytecode provider name : javassist
2017-07-27 12:48:05,346 WARN HHH000059: Defining hibernate.transaction.flush_before_completion=true ignored in HEM
2017-07-27 12:48:05,391 INFO HCANN000001: Hibernate Commons Annotations {5.0.1.Final}
2017-07-27 12:48:05,784 WARN HHH000341: Could not obtain connection metadata : SSL connection is required. Please specify SSL options and retry.
2017-07-27 12:48:05,840 INFO HHH000400: Using dialect: com.confighub.core.store.u
2017-07-27 12:48:05,861 INFO HHH000422: Disabling contextual LOB creation as connection was null
2017-07-27 12:48:05,868 INFO Envers integration enabled? : true
2017-07-27 12:48:07,136 INFO HHH000250: Starting update timestamps cache at region: org.hibernate.cache.spi.UpdateTimestampsCache
2017-07-27 12:48:07,138 WARN HHH020003: Could not find a specific ehcache configuration for cache named [org.hibernate.cache.spi.UpdateTimestampsCache]; using defaults.
2017-07-27 12:48:07,145 INFO HHH000248: Starting query cache at region: org.hibernate.cache.internal.StandardQueryCache
2017-07-27 12:48:07,145 WARN HHH020003: Could not find a specific ehcache configuration for cache named [org.hibernate.cache.internal.StandardQueryCache]; using defaults.
2017-07-27 12:48:07,539 WARN SQL Error: 2001, SQLState: 28000
2017-07-27 12:48:07,539 ERROR SSL connection is required. Please specify SSL options and retry.
2017-07-27 12:48:07,585 WARN SQL Error: 2001, SQLState: 28000
2017-07-27 12:48:07,585 ERROR SSL connection is required. Please specify SSL options and retry.
```

**Comment by edoarsla** [comment]
Hi. Are you able to connect without SSL enabled?  Which database are you using?

**Comment by shrock-n-awe** [comment]
If I disable the SSL requirement on the server I am able to connect.

With SSL disabled on the server the value of db.main.useSSL does not seem to matter, and I am able to connect with it set to true or false. With SSL required on the server side db.main.useSSL = true does not work.

I am using MySQL 5.3, from the installation docs I thought this was the only supported option, are other databases supported as well? Could I use postgres?

**Comment by edoarsla** [comment]
Not right now. We have to add Postgres drivers - which we plan to do anyway. If you need it, we can reprioritize it and push a release with that toggle. 

We'll take another look at the SSL flag and get back to you.

**Comment by shrock-n-awe** [comment]
I could use postgres with no encyrption since we still have a legacy IaaS instance. But we are moving to PaaS hosted database platforms going forward, so SSL support for DB connections is pretty important for our us.

**Comment by edoarsla** [comment]
Got it.  Both, a fix for SSL, and support for postgres are going to be in the next release.

**Comment by shrock-n-awe** [comment]
Any idea on timeframe? Also are you interested in taking pull requests/contributions? I would be willing to help out if you are interested, and have a process for that.

**Comment by edoarsla** [comment]
I believe we’re currently on track for next 2-3 weeks.
I’d be very happy to discuss the contributions, and I’m interested to learn about the process.  If you create a demo request on our website, we can set some time and discuss further?

**Comment by shrock-n-awe** [comment]
Ok... i will get something setup later tonight!

**Comment by edoarsla** [comment]
We just published v1.6 with support for PostgreSQL 9+.
This version configures database via `server/conf/tomee.xml` file and allows you to directly control all aspects of database configuration.

To specify SSL related parameters, you will use "connectionProperties" parameter.  Full documentation is here: http://tomee.apache.org/datasource-config.html#connectionProperties.

Please let us know if you have any problems.

**Comment by edoarsla** [comment]
This is fixed as of v1.6.  You may specify database specific parameters in the server/conf/tomee.xml file.   For example, using connectionProperties=[ssl=true, ...]
